client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -325,8 +409,12 @@ public boolean replace(K key, V oldValue, V value, long lifespan, TimeUnit lifes
    public CompletableFuture<V> replaceAsync(K key, V value, long lifespan, TimeUnit lifespanUnit, long maxIdleTime, TimeUnit maxIdleTimeUnit) {
       assertRemoteCacheManagerIsStarted();
       ReplaceOperation<V> op = operationsFactory.newReplaceOperation(compatKeyIfNeeded(key),
-            keyToBytes(key), valueToBytes(value), lifespan, lifespanUnit, maxIdleTime, maxIdleTimeUnit, dataFormat);
-      return op.execute();
+             keyToBytes(key), valueToBytes(value), lifespan, lifespanUnit, maxIdleTime, maxIdleTimeUnit, dataFormat);
+      CompletableFuture<V> result = op.execute();
+      if (statistics) {
+         result.thenRun(() -> clientStatistics.incrementRemoteStores());
```

Comment:
This is the trouble with cond commands we've talked about...

Reply:
My next push will move statistics collection to the operations instead of at the RemoteCacheImpl level.